web/js/app.js

```diff
@@ -530,25 +655,57 @@ LR.utils.handleMultiSelect = function(event, fromTemplate){
     input.setAttribute("value", str_arr)
 }
 
+/**
+ * A hidden input is tracking a select or multi select. It is hidden and it is not a primitive, so DEER does not handle the value.
+ * Make sure to set the value of this hidden input.  The value is a string, that string contains a delimiter to join on for cases of multiple values.
+ * 
+ * @param {Object} annotationData Expanded data that is all annotation data for a form.
+ * @param {Array(String)} keys The specific annotations we are looking for in annotationData. 
+ * @param {HTMLElement} form The completely loaded HTML <form> containing the <selects>s
+ * @return None
+ */
+LR.utils.setTrackedHiddenValues = function(annotationData, keys, form){
+    keys.forEach(key =>{
+        let data_arr = 
+        (annotationData[key].hasOwnProperty("value") && annotationData[key].value.hasOwnProperty("items")) ? annotationData[key].value.items : annotationData[key].hasOwnProperty("items") ? annotationData[key].items : [ LR.utils.getAnnoValue(annotationData[key]) ]
+        let input = form.querySelector("input[deer-key='"+key+"']")
+        //Set the value of the hidden input that tracks this for DEER
+        //Check if we need a different delimeter.  The input will tell us.
+        let delim = (input.hasAttribute("deer-array-delimeter")) ? input.getAttribute("deer-array-delimeter") : ","
+        //Generate the value for the input that DEER supports - "uri,uri..."
+        let str_arr = (data_arr.length > 1) ? data_arr.join(delim) : (data_arr.length === 1 ) ? data_arr[0] : ""
+        input.value = str_arr
+    })
+}
+
 /**
  * 
  * Make sure not to select options outside the <form> and <select> involved here.  
- * @param {Object} annotationData The expanded containing all annotation data for a form.
- * @param {Array(String)} keys The specific annotations we are looking for in annotationData
+ * This may be for a basic <select>, it may be for a <select multiple>
+ * These are dynamic selects built with custom UI and so DEER cannot preselect these.  We have to do it after the form loads.
+ * For dynamic selects, we follow a convention where there is a hidden input at this.parentElement.previousElementSibling with the deer-key
+ * Make sure to select the appropriate <options>
+ * 
+ * How about we do this with optional chaining.  This should work to pre select simple dropdowns AND multi selects (always at least an array of 1).
+ * 
+ * @param {Object} annotationData Expanded data that is all annotation data for a form.
+ * @param {Array(String)} keys The specific annotations we are looking for in annotationData. 
  * @param {HTMLElement} form The completely loaded HTML <form> containing the <selects>s
  * @return None
  */
-LR.utils.preSelectMultiSelects = function(annotationData, keys, form){
+LR.utils.preSelectSelects = function(annotationData, keys, form){
```

Comment:
Technically, this should be `preselectSelects` but that can be refactored later.